### PR TITLE
Added update, enable, disable events to the updateStoragePool API

### PR DESCRIPTION
**Pull request by harikrishna-patnala**
### Description

This PR addresses the issue #8494 

Fixes the issue of missing Enable, Disable action events when trying to do that using updateStoragePool. Also added new Action event UPDATE.PS to the actual update storage pool event.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):

Screenshot 2024-08-19 at 5 52 15 PM

### CODE REVIEW

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

File: api/src/main/java/org/apache/cloudstack/api/command/admin/storage/UpdateStoragePoolCmd.java [hunk 1]
```diff
@@ -147,7 +147,22 @@ public void setUrl(String url) {
...
     public void execute() {
-        StoragePool result = _storageService.updateStoragePool(this);
+        StoragePool result = null;
+        if (name != null ||
+                (tags != null && !tags.isEmpty()) ||
+                capacityIops != null ||
+                capacityBytes != null ||
+                (details != null && !details.isEmpty()) ||
+                url != null ||
+                isTagARule != null) {
```
Comment (JoaoJandre) [inline api/src/main/java/org/apache/cloudstack/api/command/admin/storage/UpdateStoragePoolCmd.java:157]: We could use ObjectUtils.anyNotNull() for all non collection variables and CollectionUtils.isNotEmpty() for details and tags.

Comment (harikrishna-patnala) [reply api/src/main/java/org/apache/cloudstack/api/command/admin/storage/UpdateStoragePoolCmd.java]: updated the code. Thanks @JoaoJandre

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

**Review by kiranchavala — APPROVED** [review]
LGTM

DISABLE, ENABLE, UPDATE PS events are properly generated 

Screenshot 2024-08-21 at 1 47 52 PM

**Comment by kiranchavala** [comment]
@blueorangutan test

**Review by JoaoJandre — APPROVED** [review]
CLGTM, did not test it

**Comment by harikrishna-patnala** [comment]
test_13_migrate_volume_and_change_offering failure is because of this PR change, will update the PR accordingly.

cc @DaanHoogland

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

**Comment by harikrishna-patnala** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

File: server/src/main/java/com/cloud/storage/StorageManagerImpl.java [hunk 2]
```diff
@@ -1127,8 +1127,10 @@ private Map<String, String> extractApiParamAsMap(Map ds) {
         return details;
     }
 
+    @Override
     @ActionEvent(eventType = EventTypes.EVENT_DISABLE_PRIMARY_STORAGE, eventDescription = "disable storage pool")
-    private void disablePrimaryStoragePool(StoragePoolVO primaryStorage) {
+    public StoragePool disablePrimaryStoragePool(Long id) {
+        StoragePoolVO primaryStorage = _storagePoolDao.findById(id);
         if (!primaryStorage.getStatus().equals(StoragePoolStatus.Up)) {
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/storage/StorageManagerImpl.java:1134]: @harikrishna-patnala might result in NPE on enable/disable for removed pools, it is checked in update storage pool flow earlier.

Comment (harikrishna-patnala) [reply server/src/main/java/com/cloud/storage/StorageManagerImpl.java:1137]: added the ID check @sureshanaparti

**Comment by harikrishna-patnala** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Pull request merged**